routing/bus/command.go

```
@@ -64,15 +59,10 @@ func CommandBus(router *message.Router,
 }
 
 func (h *commandBusHandler) HandleMessage(msg *message.Message) ([]*message.Message, error) {
-	cloudMessage := &routingmessage.CloudMessage{}
-	if err := json.Unmarshal(msg.Payload, cloudMessage); err != nil {
-		return nil, errors.Wrap(err, "cannot deserialize cloud message")
-	}
-	msg.SetContext(command.SetMessageToContext(msg, cloudMessage))
 	for _, commandHandler := range h.commandHandlers {
-		if util.ContainsString(commandHandler.Topics(), cloudMessage.CommandName) {
-			return commandHandler.HandleMessage(msg)
+		if msg, err := commandHandler.HandleMessage(msg); err == nil {
```

The error is ignored but should not be.